test/parallel/test-worker-track-unmanaged-fds.js

```diff
@@ -14,15 +14,14 @@ if (!isMainThread)
 const preamble = `
 const fs = require("fs");
 const { parentPort } = require('worker_threads');
-const __filename = ${JSON.stringify(__filename)};
 process.on('warning', (warning) => parentPort.postMessage({ warning }));
 `;
 
 (async () => {
   // Consistency check: Without trackUnmanagedFds, FDs are *not* closed.
   {
     const w = new Worker(`${preamble}
-    parentPort.postMessage(fs.openSync(__filename));
+    parentPort.postMessage(fs.openSync('${__filename}'));
```

Comment:
That's an unrelated change – and an unsafe one, if `__filename` contains a `'`, it would make the test throw. Please revert the changes in this file.

Reply:
this change was not voluntary and `__filename` in this case does not contain a `'`. If in the future it would, the test would immediately fail so it's imo not dangerous.
I made this change becaue with the initial `preamble` V8 does not compile cleanly in the trial compile to detect the module type triggered [here](https://github.com/dygabo/node/blob/fix-for-unflagging-module-format-detection/lib/internal/modules/esm/get_format.js#L99).
Looking a bit under the hood that happens because of `const __filename` when v8 looks at it [here](https://github.com/nodejs/node/blob/main/deps/v8/src/api/api.cc#L2747).
This causes `containsModuleSyntax` in `node_contextify.cc` to falsely think this is an ESM module (see [here](https://github.com/dygabo/node/blob/fix-for-unflagging-module-format-detection/src/node_contextify.cc#L1502)). Because the logic is: try to compile it, if it has errors, it's not CJS, so it must contains ESM syntax.

Having an `eval` module with `__filename` value is a nice trick but IMO it is exactly that: a trick. And it is one that in this case plays against us. 
For that reason I would keep this change if we don't have a better solution for it. As mentioned, the danger of `__filename` to contain sometimes in the future that `'` character in the name is relatively small and if it happens, we catch it already on `make jstest'

Reply:
> this change was not voluntary and `__filename` in this case does not contain a `'`

You don’t know that, it depends where the nodejs/node repo is cloned

Reply:
Renaming the variable would do the trick. But I think we should not do that, instead we should fix the bug – because it's clearly a bug.

Reply:
I've opened https://github.com/nodejs/node/pull/53050 to fix the regression

Reply:
https://github.com/nodejs/node/pull/53050 has landed

Reply:
this branch is now rebased to include that fix. I will work on addressing the rest of the comments and try to propose a solution that will work as good as it gets for the `resolve` hook. I don't have a plan for the transpile scenario as that information is only available during `load` but I think the rest of the cases can correctly set the module type on `resolve`.